inference/core/utils/file_system.py

```diff
@@ -0,0 +1,60 @@
+import json
+import os.path
+from typing import List, Optional, Union
+
+
+def read_text_file(
+    path: str,
+    split_lines: bool = True,
+    strip_white_chars: bool = True,
+) -> Union[str, List[str]]:
+    with open(path) as f:
+        if split_lines:
+            lines = list(f.readlines())
+            if strip_white_chars:
+                return [line.strip() for line in lines if len(line.strip()) > 0]
+        content = f.read()
+        if split_lines:
```

Comment:
Should this be `if strip_white_chars` ?

Reply:
👍  good catch, I was about to test this 😆 that's why tests are so needed